src/Donations/Routes/RegisterDonationRoutes.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Give\Donations\Routes;
+
+use Give\Donations\Controllers\DonationRequestController;
+use Give\Donations\ValueObjects\DonationRoute;
+use WP_REST_Request;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class RegisterDonationRoutes
+{
+    /**
+     * @var DonationRequestController
+     */
+    protected $donationRequestController;
+
+    /**
+     * @unreleased
+     */
+    public function __construct(DonationRequestController $donationRequestController)
+    {
+        $this->donationRequestController = $donationRequestController;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __invoke()
+    {
+        $this->registerGetDonation();
+        $this->registerGetDonations();
+    }
+
+    /**
+     * Get Donation route
+     *
+     * @unreleased
+     */
+    public function registerGetDonation()
+    {
+        register_rest_route(
+            DonationRoute::NAMESPACE,
+            DonationRoute::DONATION,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        return $this->donationRequestController->getDonation($request);
+                    },
+                    'permission_callback' => '__return_true',
```

Comment:
We need to update this one as well.

Reply:
Resolved in 0bc52d67abe9047987a915fc436ddab1cfbc91c4